src/main/java/tool/designpatterns/verifiers/multiclassverifiers/decorator/DecoratorVerifier.java

```diff
@@ -0,0 +1,226 @@
+     *
+     * @return A {@link Feedback} object containing the result
+     */
+    public Feedback interfaceContainsMethod(
+        ClassOrInterfaceDeclaration interfaceComponent) {
+        Feedback result;
+        MethodDeclaration erroringMethod = null;
+
+        AtomicBoolean resultBool = new AtomicBoolean(false);
+        for (MethodDeclaration methodDeclaration : interfaceComponent.findAll(
+            MethodDeclaration.class)) {
+            if (!methodDeclaration.isPrivate()) {
+                resultBool.set(true);
+                erroringMethod = methodDeclaration;
+            }
+        }
+
+        if (resultBool.get()) {
+            result = Feedback.getSuccessfulFeedback();
+        } else {
+            String msg = "Interface does not contain any methods!";
+            if (erroringMethod == null) {
+                result = Feedback.getPatternInstanceNoChildFeedback(msg);
+            } else {
+                result = Feedback.getNoChildFeedback(msg, new FeedbackTrace(erroringMethod));
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Method for checking that a class contains a field of a certain type.
+     *
+     * @param toTest The class to check
+     * @param type   The class or interface that toTest should contain
+     *
+     * @return A {@link Feedback} object containing the result
+     */
+    @SuppressWarnings("PMD.LinguisticNaming")
+    private Feedback hasFieldOfType(
+        ClassOrInterfaceDeclaration toTest, ClassOrInterfaceDeclaration type) {
+        Feedback result;
+        AtomicBoolean hasAComponent = new AtomicBoolean(false);
+        toTest.findAll(VariableDeclarator.class).forEach(fieldDeclaration -> {
+            if (fieldDeclaration.getTypeAsString().contains(
+                type.getNameAsString())) { //Check that this works
+                hasAComponent.set(true);
+            }
+        });
+        if (hasAComponent.get()) {
+            result = Feedback.getSuccessfulFeedback();
+        } else {
+            result = Feedback.getNoChildFeedback(
+                "There was no field with type '" + type.getNameAsString() + "'",
+                new FeedbackTrace(toTest));
+        }
+        return result;
+    }
+
+    /**
+     * Checks if all constructors in a given class initialize the Component field in the class.
+     *
+     * @param toTest        The class to check.
+     * @param interfaceName
+     *
+     * @return A {@link Feedback} object containing the result
+     */
+    private Feedback fieldInitializedInConstr(
+        ClassOrInterfaceDeclaration toTest, ClassOrInterfaceDeclaration interfaceName) {
+        Feedback result;
+        AtomicBoolean isInitialized = new AtomicBoolean(true);
+        List<FieldDeclaration> fieldsInClass = new ArrayList<>();
+        toTest.findAll(FieldDeclaration.class).forEach(fieldDeclaration -> {
+            fieldsInClass.add(fieldDeclaration);
+        });
```

Comment:
```suggestion
        List<FieldDeclaration> fieldsInClass = new ArrayList<>(
            toTest.findAll(FieldDeclaration.class));
```

Reply:
Intellij suggested this change for me and it hade some other suggestions as well. Use Inteillij well and it can simplify and shorten the code for you.

Reply:
Reasonable change, missed that findAll returns a List rather than some other type, this can probaply be applied elsewhere in the code aswell!